cub/device/device_histogram.cuh

```diff
@@ -72,10 +72,15 @@ struct DeviceHistogram
    *
    * @par
    * - The number of histogram bins is (`num_levels - 1`)
-   * - All bins comprise the same width of sample values: 
-   *   `(upper_level - lower_level) / (num_levels - 1)`
-   * - The ranges `[d_samples, d_samples + num_samples)` and 
-   *   `[d_histogram, d_histogram + num_levels - 1)` shall not overlap 
+   * - All bins comprise the same width of sample values:
+   *   `(upper_level - lower_level) / (num_levels - 1)`.
+   * - For an integral `SampleT` the bin for a sample is computed as `(sample - lower_level) *
```

Comment:
We do not define what `SampleT` is, but I guess it should be obvious.

Reply:
Strictly speaking, this is `If std::common_type<LevelT, SampleT> is integral, the bin for a sample is computed as ...`. It's more accurate but also a bit more convoluted. What's your preference?

Reply:
@elstehle good point! I'd spell that, but probably in natural language instead, something like: "If the common type of level and sample is of integral type, ..."

Reply:
Opted for the more precise formulation. Please have another look and resolve, if you're satisfied 🙂